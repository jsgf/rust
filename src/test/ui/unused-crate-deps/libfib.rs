// check-pass
// aux-crate:bar=bar.rs
// compile-flags:--crate-type lib -Wunused-crate-deps

pub fn fib(n: u32) -> Vec<u32> {
//~^ External crate `bar` unused in `libfib`. Remove the dependency or add `use bar as _;`.
let mut prev = 0;
    let mut cur = 1;
    let mut v = vec![];

    for _ in 0..n {
        v.push(prev);
        let n = prev + cur;
        prev = cur;
        cur = n;
    }

    v
}
