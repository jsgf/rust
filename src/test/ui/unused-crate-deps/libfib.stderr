warning: External crate `bar` unused in `libfib`. Remove the dependency or add `use bar as _;`.
  --> $DIR/libfib.rs:5:1
   |
LL | pub fn fib(n: u32) -> Vec<u32> {
   | ^
   |
   = note: requested on the command line with `-W unused-crate-deps`

